test/gtest/ucp/test_ucp_wakeup.cc

```diff
@@ -117,6 +117,15 @@ UCS_TEST_P(test_ucp_wakeup, tx_wait, "ZCOPY_THRESH=10000")
     std::string send_data(COUNT, '2'), recv_data(COUNT, '1');
     void *sreq, *rreq;
 
+    if (std::find(GetParam().transports.begin(),
+                  GetParam().transports.end(),
+                  "tcp") != GetParam().transports.end()) {
+        UCS_TEST_SKIP_R("This test doesn't progress receiver's worker, while "
+                        "waiting for the events on a sender's worker fd. "
+                        "So, this causes the hang due to lack of the progress "
```

Comment:
Please don't hide the root cause. The test does right thing as it designed, the hang caused by lack of functionality in TCP - async messaging on wireup